alt_e2eshark/onnx_tests/operators/tfidf_vectorizer.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+from typing import Dict, Any
+
+from onnx import TensorProto
+from onnx.helper import make_node, make_tensor_value_info
+
+from ..helper_classes import BuildAModel
+from e2e_testing.registry import register_test
+
+TEST_PARAMS = {
+    "only_bigram": {
+        "n": 2,
+        "min_gram_length": 2,
+        "max_gram_length": 2,
+        "max_skip_count": 1,
+        "ngram_counts": [0, 1],
+        "pool_int64s": [1, 2, 3, 5, 4, 5, 6, 7, 8, 6, 7],
+        "ngram_indexes": [0, 1, 2, 3, 4, 5, 6],
+        "weights": [0.5, 0.0, 1.0, 1.0, 7.0, 2.0, 0.0],
+    },
+    "unigram_bigram": {
+        "n": 4,
+        "min_gram_length": 1,
+        "max_gram_length": 2,
+        "max_skip_count": 0,
+        "ngram_counts":[0,4],
+        "pool_int64s": [2, 3, 5, 4, 5, 6, 7, 8, 6, 7],
+        "ngram_indexes": [0, 2, 4, 5, 6, 7, 8, 9],
+        "weights": [0.491, 0.15, 0.1, 1.0, 1.0, 7.0, 2.0, 0.0],
+    },
+    "unigram_bigram_skip5": {
+        "n": 3,
+        "min_gram_length": 1,
+        "max_gram_length": 2,
+        "max_skip_count": 5,
+        "ngram_counts":[0,4],
+        "pool_int64s": [2, 3, 5, 4, 5, 6, 7, 8, 6, 7],
+        "ngram_indexes": [0, 2, 4, 5, 6, 7, 8, 9],
+        "weights": [0.0, 0.15, 0.1, 1.0, 1.0, 7.0, 2.0, 0.0],
+    },
+}
+
+def create_vectorizer_node(
+        mode: str,
+        params: Dict[str, Any]
+    ):
+    return make_node(
+        op_type="TfIdfVectorizer",
+        inputs=["X"],
+        outputs=["Y"],
+        mode=mode,
+        min_gram_length=params["min_gram_length"],
+        max_gram_length=params["max_gram_length"],
+        max_skip_count=params["max_skip_count"],
+        ngram_counts=params["ngram_counts"],
+        ngram_indexes=params["ngram_indexes"],
+        pool_int64s=params["pool_int64s"],
+        weights=params["weights"],
+    )
+
+# mode=TF
+class TfIdfVectorizerIDFOnlyBigramModel(BuildAModel):
```

Comment:
It might be a bit more compact to do something like 
```python
def construct_test_info(params_str, mode):
   class TFIDFModel(BuildAModel):
      def construct_i_o_value_info(self):
         params=TEST_PARAMS[params_str]
...
   return TFIDFModel

for n in TEST_PARAMS.keys():
   register_test(construct_test_info(n, mode(maybe attach to TEST_PARAMS?)), f'tfidf_{mode}_{n}')
```